doc/datatable.md

```diff
@@ -1,27 +1,36 @@
 
 `DataTable` is react reusable component.
-It is used to display common data like numbers and strings alongside with JSON objects.
+It is used to display common data like numbers and strings. 
+
+Also `DataTable` supports `advmode`. When `advmode` is enabled, `DataTable` displays extra column which is json for data passed in as `data` prop. More info about `advmode` can be found in `doc/advmode.md`.
+
+Demo app's path: `asab-webui/demo/src/modules/home/containers/TableContainer.js`.
 
 # Implementation
 
+DataTable can be mounted in its own container as it is implemented in demo app or as a widget in another app's container. In first case you must create container for `DataTable` and connect it to Home Module of your application by adding it's route and navigation item. Example may be found in `demo/src/modules/home/index.js`.
+
 DataTable needs obligatory props `headers`, `data`, `count`, `currentPage`, `setPage` and `title` to render.
 
+Some props can be obtained from configuration of the project  (e.g. in demo app `headers` and `limit` props are obtained from the configuration).
+
 Prop `title` is an object that has obligatory property `text` which is string and it will be rendered as table title. Also `title` has optional property `icon` for rendering icon alongside the table title and which can be either React Component (e.g. from package `@material-ui/icons`) or it can also be a string which is classname for icon element and will be rendered as <i className={icon}></i>
 
-Prop `headers` is basically an array containing objects with obligatory properties `text` and `type`. Property `text` is a string that will be rendered as a header (first letter of `headers.text` automaticly will be converted to uppercase) in headers row of the table. Property `type` is a type of data in each cell for column of that header, it can be either `string` or `json`.
-Order of headers in a table is the same as in prop `headers`.
+Prop `headers` is basically an array containing objects with obligatory properties `name` and `key`. Property `name` is a string that will be rendered as a header cell in headers row of the table. Property `key` is a key name for getting data from `data` prop, it must be the same as it is in objects in `data` prop.
+Order of headers in a table is the same as it is in prop `headers`.
 
-Prop `data` is an array of objects. `DataTable` goes through `data` and looks for those properties which has the same name as 
-`headers[idx].text`, if such exist, it will be rendered under that `headers[idx].text` as string or json.
+Prop `data` is an array of objects. `DataTable` component goes through objects in `data` and for each object looks for those properties which has the same key name that was defined in `headers[idx]`. If properties exist in that object from `data`, they will be rendered as string under the appropriate headers. Otherwise if some properties doesn't exist in that object, the `-` symbol will be rendered instead.
```

Comment:
@titkov49 I would write here, that data are usually obtained from API endpoint with the use of axios and add there a simple example, e.g:

```
   ...
	const retrieveData = () => {
		Axios.get()
		.then(response => {
			setData(response.data);
		})
		.catch(e => {
			console.log(e);
			props.app.addAlert("warning", "Failed to fetch the data");
		});
	};
   ...
```